sky/serve/replica_managers.py

```diff
@@ -1205,7 +1205,22 @@ def update_version(self, version: int, spec: 'service_spec.SkyServiceSpec',
                 for key in ['service']:
                     old_config.pop(key)
                 # Bump replica version if all fields except for service are
-                # the same. File mounts should both be empty, as update always
+                # the same.
+                # Here, we manually convert the any_of field to a set to avoid
+                # only the difference in the random order of the any_of fields.
+                old_config_any_of = old_config.get('resources',
+                                                   {}).pop('any_of', None)
+                new_config_any_of = new_config.get('resources',
+                                                   {}).pop('any_of', None)
+                if old_config_any_of is not None and new_config_any_of is None:
+                    continue
+                if old_config_any_of is None and new_config_any_of is not None:
+                    continue
+                if (old_config_any_of is not None and
+                        new_config_any_of is not None):
+                    if set(old_config_any_of) != set(new_config_any_of):
+                        continue
```

Comment:
I'm wondering if the branches could be reduced by setting the default value of `.pop` to something that can be casted as a set, like this:
```suggestion
                old_config_any_of = set(old_config.get('resources',
                                                   {}).pop('any_of', {}))
                new_config_any_of = set(new_config.get('resources',
                                                   {}).pop('any_of', {}))
                if old_config_any_of != new_config_any_of:
                        continue
```